src/map.cpp

```diff
@@ -5222,7 +5222,8 @@ void map::load(const int wx, const int wy, const int wz, const bool update_vehic
 
 void map::shift_traps( const tripoint &shift )
 {
-    const tripoint offset( shift.x * SEEX, shift.y * SEEY, shift.z );
+    // Offset needs to have sign opposite to shift direction
+    const tripoint offset( -shift.x * SEEX, -shift.y * SEEY, -shift.z );
```

Comment:
Good find, I should have tested this more carefully.

~~Why didn't you not simply change the `pos += offset` to `pos -= offset`? Seems simpler.~~ Never mind, there is currently no `operator-` for tripoints.